compiler/frontend/spark/src/main/java/edu/snu/coral/compiler/frontend/spark/source/SparkBoundedSourceVertex.java

```diff
@@ -60,25 +62,34 @@ public SparkBoundedSourceVertex getClone() {
    * A Readable for SparkBoundedSourceReadablesWrapper.
    */
   private final class SparkBoundedSourceReadable implements Readable<T> {
-    private final SparkConf conf;
-    private final Iterable<T> iterable;
+    private final LinkedHashMap<String, Object[]> commands;
+    private final Map<String, String> sessionInitialConf;
+    private final int partitionIndex;
 
     /**
      * Constructor.
-     * @param partition partition for this readable.
-     * @param rdd rdd to read data from.
+     * @param commands list of commands needed to build the dataset.
+     * @param sessionInitialConf spark session's initial configuration.
+     * @param partitionIndex partition for this readable.
      */
-    private SparkBoundedSourceReadable(final Partition partition, final RDD<T> rdd) {
-      this.conf = rdd.sparkContext().conf();
-      // TODO #756: make this bit distributed.
-      this.iterable = Lists.newArrayList(() ->
-          JavaConverters.asJavaIteratorConverter(rdd.iterator(partition, TaskContext$.MODULE$.empty())).asJava());
+    private SparkBoundedSourceReadable(final LinkedHashMap<String, Object[]> commands,
+                                       final Map<String, String> sessionInitialConf,
+                                       final int partitionIndex) {
+      this.commands = commands;
+      this.sessionInitialConf = sessionInitialConf;
+      this.partitionIndex = partitionIndex;
     }
 
     @Override
-    public Iterable<T> read() {
-      new SparkContext(conf);
-      return iterable;
+    public Iterable<T> read() throws Exception {
+      final SparkSession spark = SparkSession.builder()
+          .config(sessionInitialConf)
+          .getOrCreate();
+      final Dataset<T> dataset = SparkSession.initializeDataset(spark, commands);
+
+      final RDD<T> rdd = dataset.rdd();
```

Comment:
Could you add some comments that this operation does not load the full dataset but only the handler of partitions?

Reply:
Sure. I'll do that.